docs/types/dynamic-dispatch.md

```diff
@@ -162,3 +182,54 @@ main = Test.simplify
 When invoking a method on _module object_ its _module static methods_ take
 precedence over _instance methods_ defined on `Any`. Thus a module serves
 primarily as a _container for module (static) methods_.
+
+### Diagram of method resolution and invocation on `Any`
+
+This section describes the _special_ method resolution and invocation on `Any`
+type. More specifically, it describes when a `self` argument is implicitly
+provided and what kind of `self` it is.
+
+Considering:
+
+```
+@Builtin_Type
+type Any
+    to_text self = ...
+    ...
+
+type T
+    method self = "T.method:" + self.to_text
+
+Any.method self = "Any.method:" + self.to_text
+```
+
+where `T.method` "overrides" `Any.method`, the method resolution and invocation
+algorithm can be generally described as follows:
+
+- Is the method called statically? For example like `Any.method ...` or
+  `T.method ...`.
```

Comment:
How does one find out that the "method is called statically"? By receiver being a `Type`?

Reply:
Yes. Added this sentence to docs in e426eafbd39afd88365318ba38b4a7fb059b1495